kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/AbstractRefinableView.java

```diff
@@ -223,4 +223,41 @@ public URI getUri() {
     }
     return builder.build();
   }
+
+  @Override
+  public void signalReady() {
+    Pair<DatasetRepository, Map<String, String>> pair =
+        Registration.lookupDatasetUri(URI.create(getUri().getRawSchemeSpecificPart()));
+    DatasetRepository repo = pair.first();
+
+    repo.signalReady(dataset.getNamespace(), dataset.getName(), getConstraints());
+  }
+
+  @Override
+  public boolean isReady() {
+    Pair<DatasetRepository, Map<String, String>> pair =
+        Registration.lookupDatasetUri(URI.create(getUri().getRawSchemeSpecificPart()));
+    DatasetRepository repo = pair.first();
```

Comment:
Currently the ready signals are stored with the MetadataProvider. There didn't seem to be any direct way to determine the metadata provider for a Dataset (although I certainly might have missed it). My first pass exposed the metadata provider through the repository for the dataset to work against but it felt cleaner to work with the repository rather than directly with the metadata provider.